packages/hierarchies-react/src/presentation-hierarchies-react/itwinui/RootErrorRenderer.tsx

```diff
@@ -0,0 +1,89 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+import { ReactElement } from "react";
+import { Button, Icon, Text } from "@itwin/itwinui-react/bricks";
+import { useTree } from "../UseTree.js";
+import { ErrorNode } from "./FlatTreeNode.js";
+import { useLocalizationContext } from "./LocalizationContext.js";
+
+const errorSvg = new URL("@itwin/itwinui-icons/status-error.svg", import.meta.url).href;
+
+/**
+ * @alpha
+ */
+export type RootErrorRendererProps = {
+  /** Root error to be displayed */
+  errorNode: ErrorNode;
+  // Callback to render custom root errors.
+  renderRootError?: ({ errorNode }: { errorNode: ErrorNode }) => ReactElement;
+} & Pick<ReturnType<typeof useTree>, "getHierarchyLevelDetails" | "reloadTree">;
+
+/**
+ * A component that renders root node error state.
+ *
+ * @alpha
+ */
+export function RootErrorRenderer({ errorNode, renderRootError, getHierarchyLevelDetails, reloadTree }: RootErrorRendererProps) {
+  const { localizedStrings } = useLocalizationContext();
+
+  if (renderRootError) {
+    return renderRootError({ errorNode });
+  }
```

Comment:
What's the point of having `renderRootError`? It seems whoever would pass that prop could simply completely skip `RootErrorRenderer` and just render their own component. Or am I missing something?

Reply:
I wanted to keep the rootErrorRenderer as close as errorRenderer so in a way you could provide the same custom component to both.

Reply:
Wouldn't styling be broken if you pass component used in `renderError` to the `rootErrorRenderer.renderError`? Because in case of `ErrorRenderer` it used that component for each error item, while in `RootErrorRenderer` only that custom component is rendered.

Reply:
I just don't see any value in it... From your example:

```ts
<TreeRenderer
    {...treeProps}
    rootErrorRenderer={(rootErrorRendererProps) => (
        <RootErrorRenderer
            {...rootErrorRendererProps}
            renderRootError={(rootErrorProps) => {
                return <>My error</>;
            }}
        />
    )}
/>
```

You could instead just do this to get the same result:

```ts
<TreeRenderer
    {...treeProps}
    rootErrorRenderer={(rootErrorRendererProps) => {
        return <>My error</>;
    }}
/>
```

Reply:
Removed `renderError` from `rootErrorRenderer`.